openid4vc-high-assurance-interoperability-profile-1_0.md

```diff
@@ -231,7 +231,9 @@ The following requirements apply for both, the Wallet and the Verifier, unless s
   * The Wallet MUST support both signed and unsigned requests as defined in Annex A.3.1 and A.3.2 of [@!OIDF.OID4VP]. The Verifier MAY support signed requests, unsigned requests, or both.
 * Wallet Invocation is done via the W3C Digital Credentials API or an equivalent platform API. Any other mechanism, including Custom URL schemes, MUST NOT be used.
 * Response Mode MUST be `dc_api.jwt`. The response MUST be encrypted.
-* Specific requirements for the response encryption are tbd (https://github.com/openid/oid4vc-haip/issues/131).
+* Response encryption MUST be performed as specified in [@!OIDF.OID4VP, section 7.3]. The JWE `alg` (algorithm) header parameter (see [@!RFC7516, section 4.1.1])
+  value MUST be `ECDH-ES` (as defined in [@!RFC7518, section 4.6]), with key agreement utilizing keys on the `P-256` curve (see [@!RFC7518, section 6.2.1.1]).
+  The JWE `enc` (encryption algorithm) header parameter (see [@!RFC7516, section 4.1.2]) value MUST be `A128GCM` (as defined in [@!RFC7518, section 5.3]).
```

Comment:
Is there a specific reason to use 128-bit keys, otherwise I'd use 256-bit keys for stronger security/compliance reasons.
```suggestion
  The JWE `enc` (encryption algorithm) header parameter (see [@!RFC7516, section 4.1.2]) value MUST be `A256GCM` (as defined in [@!RFC7518, section 5.3]).
```

Reply:
+1. good catch Oliver

Reply:
I was going for bit strength parody with P-256 ECDH. Also some ISO document that I probably shouldn't have but was somehow on my hard drive was using AES-128-GCM as the AEAD part of HPKE. So I figured if it was good enough for use with the magnificence of HPKE, then it was probably good enough for this.

Reply:
You are right, ISO 18013-7 Annex C uses AES 128 GCM while ISO 18013-5 uses AES 256 GCM. I guess this is fine then.

Reply:
@bc-pi you can have those ISO documents because you are an OIDF member and ISO and OIDF have a liaison.